### PR TITLE
update: codegen for Ruby(Faraday)

### LINKED ISSUE #147: Add CodeGen for Ruby (faraday)
**Tell us about the task you want to perform and are unable to do so because the feature is not available**

Add CodeGen for Ruby
Library - https://github.com/lostisland/faraday

### PR DESCRIPTION
**Pull request by nikkivirtuoso**
## PR Description
this PR has Ruby(faraday) codegen part

## Related Issues

- Related Issue #147 
- Closes #147 

### Checklist
- [x] I have gone through the [contributing guide](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md)
- [x] I have run the tests (`flutter test`) and all tests are passing

## Added/updated tests?
_We encourage you to add relevant test cases._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests have not been included_

### CODE REVIEW

**Comment by nikkivirtuoso** [comment]
Screenshot 2024-03-15 at 10 52 17 PM
all test cases have passed

**Comment by ashitaprasad** [comment]
Did you check that the generated codes for all the test cases are running perfectly on your local PC ?

**Comment by nikkivirtuoso** [comment]
Yes ma'am they are working you can check

File: lib/consts.dart [hunk 1]
```diff
@@ -281,6 +281,7 @@ enum CodegenLanguage {
   javaAsyncHttpClient("Java (asynchttpclient)", "java", "java"),
   javaHttpClient("Java (HttpClient)", "java", "java"),
   juliaHttp("Julia (HTTP)", "julia", "jl"),
+  rubyFaraday("Ruby(Faraday)", "Ruby", "rb");
```
Comment (ashitaprasad) [inline lib/consts.dart:284]: There should be a comma here instead of a semicolon.

**Comment by ashitaprasad** [comment]
The code has compilation errors. Please ensure that your code has no compilation errors when you raise a PR.

**Comment by ashitaprasad** [comment]
The approach used is not correct. Please look at js or go codegens to see how multipart requests are created. 
Also, the code generated for test cases POST 6 to POST 9 are not correct and giving errors, which is not happening when you send a request through API Dash.
Closing this PR, please open a new PR when you have fixed all the above issues.

**Comment by nikkivirtuoso** [comment]
@ashitaprasad ma'am can you please provide a platform where i can check the ruby code compilation as I am facing problem while using faraday lib on my laptop

**Comment by animator** [comment]
@nikkivirtuoso what erro are you getting?

**Comment by nikkivirtuoso** [comment]
On VS Code using rubyLSR extension it makes the ruby file executable along with this the faraday library I am trying to install using gem install faraday it is showing error

**Pull request closed without merging**